sound/soc/sof/intel/hda-codec.c

```diff
@@ -39,54 +39,36 @@ static void hda_codec_load_module(struct hda_codec *codec) {}
 #endif
 
 /* check jack status after resuming from suspend mode */
-void hda_codec_jack_check(struct snd_sof_dev *sdev, int status)
+void hda_codec_jack_check(struct snd_sof_dev *sdev)
 {
-	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
 	struct hda_bus *hbus = sof_to_hbus(sdev);
 	struct hdac_bus *bus = sof_to_bus(sdev);
 	struct hda_codec *codec;
-	int mask;
-
-	/*
-	 * there are two reasons for runtime resume
-	 * (1) waken up by interrupt triggered by WAKEEN feature
-	 * (2) waken up by pm get functions for some audio operations
-	 * For case (1), the bits in status mean which codec triggers
-	 * the interrupt and jacks will be checked on these codecs.
-	 * For case (2), we need to check all the non-hdmi codecs for some
-	 * cases like playback with HDMI or capture with DMIC. In these
-	 * cases, only controller is active and codecs are suspended, so
-	 * codecs can't send unsolicited event to controller. The jack polling
-	 * operation will activate codecs and unsolicited event can work
-	 * even codecs become suspended later.
-	 */
-	mask = status ? status : hda->hda_codec_mask;
-
```

Comment:
don't check status ?  If there is one codec,  there is no different for this change. But if there are two codecs, the checking will save one jack polling operation.

Reply:
but do we want that? I thought that actually would be bad? If there are 2 codecs and one of them has a bit set in the status, we will only check it, so the other codec will stay suspended? And if headphones are plugged into it, they won't be detected?

Reply:
Good question: one popular design: two codecs, one for headphone, anther one with amplifier for speaker. Some vendors may add a special codec for DMIC. 

I don't insist on checking status. It is a just optional optimization.

Reply:
On my PC I have 8 3.5 mm jack ports - 2 on the front panel and 6 on the back. They aren't all handled by the same codec, are they? How common are designs with multiple codecs providing multiple jack connectors?

Reply:
for HDaudio designs, it's typically to have an all-in-one integrated codec which can deal with amplification and jack detection. I don't think we have a need to handle jack detection across multiple HDaudio codecs, it's difficult enough for a single codec to get things right.

Reply:
yes, but there are still two codecs cases. I also talked to @bardliao and conformed this case. Takashi also had a UCM patch for two codecs.  But I think this case it is not a common case and we haven't test SOF on two codecs device.

Reply:
> yes, but there are still two codecs cases. I also talked to @bardliao and conformed this case. 

@RanderWang Sorry I thought we were talking about **I2S codec**. I know that there are some designs with more than one codec. E.g. one for headset and the other for speaker. 

> we haven't test SOF on two codecs device.

Yes, we have.